src/App.js

```diff
@@ -1,16 +1,55 @@
 import './App.css';
+import { useState } from 'react';
 import { NavigationRoutes } from './routes/Routes';
-import { Navbar } from "./components/Navigation/NavBar"
-import { Footer } from "./components/Footer/Footer"
+import { Navbar } from './components/Navigation/NavBar';
+import { Footer } from './components/Footer/Footer';
+
+import { Sidebar } from './components/Sidebar/Sidebar';
+import { useLocation } from 'react-router-dom';
 
 function App() {
+    const [sidebar, setSidebar] = useState('sidebar-container');
+    const [button, setButton] = useState('sidebar-bottom-up');
+    const [buttonText, setButtonText] = useState('Menu');
+
+    const toggleSideBar = () => {
+        setButton(() =>
+            button === 'sidebar-bottom-up' ? 'sidebar-bottom-up-active' : 'sidebar-bottom-up',
+        );
+
+        setButtonText(() => (buttonText === 'Menu' ? 'Close' : 'Menu'));
+
+        setSidebar(() =>
+            sidebar === 'sidebar-container' ? 'sidebar-active' : 'sidebar-container',
+        );
+    };
+
+    const location = useLocation();
```

Comment:
If you only need `pathname` you can destructure it right here.

Reply:
Done. Thanks!